src/component/TodoItem.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import { useTodos } from '../utils/TodoContext';
+import { Errors } from '../types/ErrorsTodo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    isLoading,
+    setErrorMessage,
+    modifiedTodoId,
+    deleteTodo,
+    updateTodo,
+    handleCompleted,
+    setIsLoading,
+  } = useTodos();
+  const { id, title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [updatedTitle, setUpdatedTitle] = useState(title);
+  const titleRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (!isLoading && isEditing && titleRef.current) {
+      titleRef.current?.focus();
+    }
+  }, [isEditing, isLoading]);
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+    const trimmedTitle = updatedTitle.trim();
+
+    if (trimmedTitle === title || !trimmedTitle) {
+      if (!trimmedTitle) {
+        await deleteTodo(id);
+      }
+
+      setIsEditing(false);
+      setUpdatedTitle(title);
+      setIsLoading(false);
+
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+      await updateTodo({ ...todo, title: trimmedTitle });
+      setIsEditing(false);
+    } catch {
+      setErrorMessage(Errors.UpdateTodo);
+      if (titleRef.current) {
+        titleRef.current.focus();
+      }
+    } finally {
+      setIsLoading(false);
+    }
```

Comment:
for fix this [comment](https://github.com/mate-academy/react_todo-app-with-api/pull/1227#pullrequestreview-2028663523) do not set isLoading to false in finally (this block work in any case)